.github/workflows/deploy.yml

```diff
@@ -1,71 +1,94 @@
-name: deploy
+name: Infrastructure CI/CD
 
 permissions:
   id-token: write  # required for requesting the JWT
   contents: read  # required for actions/checkout
 
 on:
+  pull_request:
+    branches: [ main ]
   workflow_dispatch:
     inputs:
       deployment_environment:
-        description: 'environment'
+        description: 'Environment to deploy to'
         required: true
+        type: choice
+        options:
+          - dev
+          - test
 
 jobs:
-  build_and_deploy:
+  cdk_operations:
+    name: "CDK Operations"
     runs-on: ubuntu-latest
-    environment: "${{ github.event.inputs.deployment_environment }}"
+    
+    environment: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.deployment_environment || 'synthtest' }}
     
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
 
-    - name: Checkout repository
-      uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm ci
 
-    - name: Set up node
-      uses: actions/setup-node@v2
-      with:
-        node-version: 20
+      # Only assume role for actual deployments, not for synthtest
+      - name: Assume Github OIDC role
+        uses: aws-actions/configure-aws-credentials@v4
+        if: github.event_name == 'workflow_dispatch'
+        with:
+          aws-region: us-west-2
+          role-to-assume: ${{ vars.DEPLOY_ROLE }}
+          role-session-name: MAAP-eoapi-${{ github.event.inputs.deployment_environment || 'synthtest' }}-deploy
 
-    - name: Assume Github OIDC role
-      uses: aws-actions/configure-aws-credentials@v2
-      with:
-        aws-region: us-west-2
-        role-to-assume: ${{ vars.DEPLOY_ROLE }}
-        role-session-name: MAAP-eoapi-${{ github.event.inputs.deployment_environment }}-deploy
+      - name: Import stacks variables to github output
+        id: import-stacks-vars-to-output
+        if: github.event_name == 'workflow_dispatch'
+        env:
+          AUTH_STACK_NAME: ${{ vars.AUTH_STACK_NAME }}
+          ROLES_STACK_NAME: ${{ vars.ROLES_STACK_NAME }}
+        run: bash .github/workflows/scripts/import_stacks_variables_to_github_output.sh $AUTH_STACK_NAME $ROLES_STACK_NAME
 
-    - name: Install deployment dependencies
-      run: | 
-        npm install
+      - name: Set environment variables
+        id: set-env
+        run: |
+          echo "AUTHOR=${{ github.actor }}" >> $GITHUB_ENV
```

Comment:
what's the difference between loading the variables into the github env vs using `env`?

Reply:
This way I was able to recycle the env vars across multiple steps. Although we might be able to set them at the job-level.

Reply:
I switched it so the env vars are defined at the job-level in the typical way